shoop/front/middleware.py

```diff
@@ -60,6 +62,8 @@ class ShoopFrontMiddleware(object):
 
     def process_request(self, request):
         self._set_shop(request)
+        if request.shop.maintenance_mode and not request.user.is_superuser:
+            return HttpResponse(loader.render_to_string("shoop/front/maintenance.jinja", request=request))
```

Comment:
On that note, it should be done after all of the other `_set`s, to ensure the request is in a decent state.